### DISCUSSION TITLE
BUG uuid "binary" as a PK [Aug 05]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#6170] [Discord Threads]
> 🙋‍♂️ **victorminemu**  
#### We use binary encoded uuid as our PK.  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1125781024417263717/image.png)  
![image1](https://cdn.discordapp.com/attachments/661905455894888493/1125781024727646268/image.png)

**Comment by o1lab** [comment]
> 🙋‍♂️ **victorminemu**  
#### In previous versions it some what worked "on the view it shows the relations but fails to load the entity"  
#  
> 👨‍🦲 **dstala**  
#### Can you share more details about project setup? You can copy `Projects Info` from projects menu  
#  
> 🙋‍♂️ **victorminemu**  
#### Sure <@857998597474484295> :   
#### Node: **v18.16.1**   
#### Arch: **x64**   
#### Platform: **linux**   
#### Docker: **false**   
#### RootDB: **sqlite3**   
#### PackageVersion: **0.109.3**  
#  
> 🙋‍♂️ **victorminemu**  
#### The relationships are working because i can add lookup columns  
![image0](https://cdn.discordapp.com/attachments/1125781024928964689/1126142938062266529/image.png)  
#  
> 🙋‍♂️ **victorminemu**  
#### Ok so if the relations are working, then the problem is in the GET endpoint. Maybe is trying to find the key as string or binary from the string instead of transforming the uuid to binary buffer  
![image0](https://cdn.discordapp.com/attachments/1125781024928964689/1126148315394883644/image.png)  
#  
> 🙋‍♂️ **victorminemu**  
#### I've created a new row and the uuid is correctly stored as binary, the row doesn't have any relations and still gives the error when I try to open the row.  
#  
> 👨‍🦲 **dstala**  
#### this is something that we haven't tested. can you help log the details in github issue here   
#### https://github.com/nocodb/nocodb/issues  
#  
> 🙋‍♂️ **victorminemu**  
#### Sure  
#  
> 🙋‍♂️ **victorminemu**  
#### <@857998597474484295> Got it working. Added a check on the Where constructor to check if is a uuid and transform it to a Buffer.  
![image0](https://cdn.discordapp.com/attachments/1125781024928964689/1126475348322824322/image.png)  
#  
> 🙋‍♂️ **victorminemu**  
#### I found another bug, the date filter is not working with timestamps:  
![image0](https://cdn.discordapp.com/attachments/1125781024928964689/1126480659830026390/image.png)  
#  
> 🙋‍♂️ **victorminemu**  
#### The sort works as intended  
#  
> 🙋‍♂️ **victorminemu**  
#### Just to confirm this is not caused by the fix i've made for the UUID  
#  
> 🙋‍♂️ **victorminemu**  
#### Solved  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1125781024928964689)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
